liesel/model/nodes.py

```diff
@@ -1386,6 +1542,92 @@ def __repr__(self) -> str:
         return f'{type(self).__name__}(name="{self.name}")'
 
 
+def _transform_var_with_bijector_instance(var: Var, bijector_inst: jb.Bijector) -> Var:
+    if var.dist_node is None:  # type: ignore
+        raise RuntimeError(f"{var} has no distribution")
+    InputDist = var.dist_node.distribution
+    inputs = var.dist_node.inputs
+    kwinputs = var.dist_node.kwinputs
+
+    bijector_inv = jb.Invert(bijector_inst)
+
+    def transform_dist(*args, **kwargs):
+        return jd.TransformedDistribution(InputDist(*args, **kwargs), bijector_inv)
+
+    transformed_dist = Dist(
+        transform_dist,
+        *inputs,
+        **kwinputs,
+        _name="",
+        _needs_seed=var.dist_node.needs_seed,
+    )
+
+    transformed_dist.per_obs = var.dist_node.per_obs
+
+    transformed_var = Var(
+        bijector_inv.forward(var.value),
+        transformed_dist,
+        name=f"{var.name}_transformed",
+    )
+
+    var.value_node = Calc(bijector_inst.forward, transformed_var)
+    return transformed_var
+
+
+def _transform_var_with_bijector_class(
```

Comment:
to confirm my understanding: we cannot just instanciate  the bijector here and then use `_transform_var_with_bijector_instance` because 

1) the inputs may not be available at this time
2) the bijector may change when the inputs to the bijector change

Reply:
Yes, and I think point two is the critical one: Even of the input values were available at the time that `.transform` is called, we do not want to use their concrete value here. Their values may change, and as a result, the bijector's behavior may change. So when a bijector class is used, we have to set up a wrapping system that initializes the bijector anew during every update with the current input values.